code/modules/mob/living/silicon/robot/robot_defines.dm

```diff
@@ -130,6 +133,8 @@
 	var/hasExpanded = FALSE
 	var/obj/item/hat
 	var/hat_offset = -3
+	var/hat_offset_x = list()
+	var/hat_offset_y = list()
```

Comment:
Do you have *any* use case in which the y offset would be different between dirs? I think only the x would really vary.

Reply:
The north/south sprite can have different y offset if there's a shift in perspective or something.

Reply:
Changing offset vertically looks bad in practice, though, like it's misaligned